lib/internal/modules/helpers.js

```diff
@@ -372,17 +373,22 @@ function stripTypeScriptTypes(source, filename) {
     sourceMap: sourceMapEnabled,
     filename,
   };
-  const { code, map } = parse(source, options);
-  if (map) {
-    // TODO(@marco-ippolito) When Buffer.transcode supports utf8 to
-    // base64 transformation, we should change this line.
-    const base64SourceMap = Buffer.from(map).toString('base64');
-    return `${code}\n\n//# sourceMappingURL=data:application/json;base64,${base64SourceMap}`;
+  try {
+    const { code, map } = parse(source, options);
+    if (map) {
+      // TODO(@marco-ippolito) When Buffer.transcode supports utf8 to
+      // base64 transformation, we should change this line.
+      const base64SourceMap = Buffer.from(map).toString('base64');
+      return `${code}\n\n//# sourceMappingURL=data:application/json;base64,${base64SourceMap}`;
+    }
+    // Source map is not necessary in strip-only mode. However, to map the source
+    // file in debuggers to the original TypeScript source, add a sourceURL magic
+    // comment to hint that it is a generated source.
+    return `${code}\n\n//# sourceURL=${filename}`;
+  } catch (error) {
+    // Rethrow the SWC error to provide proper error and stacktrace.
+    throw new ERR_INVALID_TYPESCRIPT_SYNTAX(error);
```

Comment:
I think we should try/catch only the call to `parse` – maybe moving it to a separate function to improve readability.

```js
function tryParse(source, options) {
  try {
    return parse(source, options);
  } catch (error) {
    // Rethrow the SWC error to provide proper error and stacktrace.
    throw new ERR_INVALID_TYPESCRIPT_SYNTAX(error);
  }
```